### ISSUE TITLE
Delete trap on nested proxies.

### ISSUE THREAD

**Issue by jackw-** [tvcutsem/harmony-reflect#46]
I am having an issue when using the delete operator on an object wrapped using nested proxies. An example program is:

```javascript
var Reflect = require("harmony-reflect");

var handler = {
    deleteProperty: function(target,name,receiver) {
        return Reflect.deleteProperty(target,name,receiver);
    }
};

var o = {x: 1, y: 2};
var inner = new Proxy(o, handler);
var outer = new Proxy(inner, handler);

delete outer.x;
```

Which in turn produces this error trace:
```
/Users/jackw/Developer/harmony-test/node_modules/harmony-reflect/reflect.js:1717
      return handler.deleteProperty(target, name);
                     ^
TypeError: undefined is not a function
    at Object.global.Reflect.deleteProperty (/Users/jackw/Developer/harmony-test/node_modules/harmony-reflect/reflect.js:1717:22)
    at Object.handler.deleteProperty (/Users/jackw/Developer/harmony-test/test.js:5:24)
    at Object.Validator [as delete] (/Users/jackw/Developer/harmony-test/node_modules/harmony-reflect/reflect.js:818:20)
    at Object.<anonymous> (/Users/jackw/Developer/harmony-test/test.js:13:8)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
```
Assuming this is a bug and not an incorrect example on my part, I believe the issue may be related to the difference in naming between the Validator trap `delete` and the Reflect function `deleteProperty`.

**Comment by tvcutsem** [comment]
Thanks for reporting. Bug should be fixed in v1.1.2 of the npm package.

**Comment by tvcutsem** [comment]
Unrelated to the bug: note that defineProperty takes only two arguments (target, name), not three as suggested by your test case.
